src/lib.rs

```diff
@@ -0,0 +1,220 @@
+extern crate regex;
+
+use regex::Regex;
+
+#[macro_use]
+use std::fs::File;
+use std::io::prelude::*;
+use std::collections::HashMap;
+
+#[derive(Debug, PartialEq)]
+pub enum IdentType {
```

Comment:
Idiomatically people usually use the word "Kind" instead of "Type" in Rust for this sort of thing, just to avoid ambiguity with types.